core/src/main/java/com/nisovin/magicspells/util/handlers/RecipeHandler.java

```diff
@@ -0,0 +1,208 @@
+package com.nisovin.magicspells.util.handlers;
+
+import java.util.Set;
+import java.util.List;
+import java.util.HashSet;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.inventory.*;
+import org.bukkit.NamespacedKey;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Listener;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.configuration.ConfigurationSection;
+
+import com.nisovin.magicspells.util.Util;
+import com.nisovin.magicspells.MagicSpells;
+
+public class RecipeHandler implements Listener {
+
+    private static Set<NamespacedKey> namespaces = new HashSet<>();
+
+    public RecipeHandler() {
+        Bukkit.getPluginManager().registerEvents(this, MagicSpells.plugin);
+    }
+
+    public static void create(ConfigurationSection config) {
```

Comment:
This method contains a lot of duplicate code. Also, it feels rather excessively long, although I'm going to give the benefit of the doubt for now that it is just due to the mentioned duplicate code.